src/apps/sections/services.py

```diff
@@ -0,0 +1,168 @@
+from decimal import Decimal
+
+from sqlalchemy import delete, select, update
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from src.apps.sections.models import Section
+from src.apps.sections.schemas import (
+    SectionInputSchema,
+    SectionOutputSchema,
+    SectionBaseOutputSchema,
+    SectionUpdateSchema
+)
+from src.core.exceptions import (
+    AlreadyExists, DoesNotExist, IsOccupied,
+    WarehouseAlreadyExistsException,
+    WarehouseDoesNotExistException,
+    SectionIsNotEmptyException,
+    NotEnoughWarehouseResourcesException,
+    TooLittleRacksAmountException,
+    TooLittleWeightAmountException
+)
+from src.apps.warehouse.services import get_all_warehouses, manage_warehouse_state
+from src.apps.warehouse.models import Warehouse
+from src.core.pagination.models import PageParams
+from src.core.pagination.schemas import PagedResponseSchema
+from src.core.pagination.services import paginate
+from src.core.utils.orm import if_exists
+
+
+async def create_section(
+    session: AsyncSession, section_input: SectionInputSchema
+) -> SectionOutputSchema:
+    section_data = section_input.dict()
+
+    warehouses = await get_all_warehouses(session, PageParams())
+    if not warehouses.total:
+        raise WarehouseDoesNotExistException
+    
+    warehouse = await if_exists(Warehouse, "id", warehouses.results[0].id, session)
+    if not warehouse.available_sections:
+        raise NotEnoughWarehouseResourcesException(resource="sections")
+    
+    section_data["warehouse_id"] = warehouse.id
+
+    new_section = Section(**section_data)
+    session.add(new_section)
+    
+    await manage_warehouse_state(
+        warehouse, adding_resources_to_warehouse=False, sections_involved=True
+    )
+    session.add(warehouse)
+    await session.commit()
+
+    return SectionOutputSchema.from_orm(new_section)
+
+
+async def get_single_section(
+    session: AsyncSession, section_id: str
+) -> SectionOutputSchema:
+    if not (section_object := await if_exists(Section, "id", section_id, session)):
+        raise DoesNotExist(Section.__name__, "id", section_id)
+
+    return SectionOutputSchema.from_orm(section_object)
```

Comment:
in relation to the previous comment, enable retrieving single section with limited amount of data